packages/@lwc/engine/src/framework/def.ts

```diff
@@ -101,21 +99,17 @@ function createComponentDef(
 
     const { name } = meta;
     let { template } = meta;
-
-    let decoratorsMeta = getDecoratorsRegisteredMeta(Ctor);
-
-    // TODO: #1295 - refactor tests that are using this declaration manually
-    if (isUndefined(decoratorsMeta)) {
-        registerDecorators(Ctor, {
-            publicMethods: getOwnValue(Ctor, 'publicMethods'),
-            publicProps: getOwnValue(Ctor, 'publicProps'),
-            track: getOwnValue(Ctor, 'track'),
-            wire: getOwnValue(Ctor, 'wire'),
-        });
-        decoratorsMeta = getDecoratorsRegisteredMeta(Ctor);
+    const decoratorsMeta = getDecoratorsRegisteredMeta(Ctor);
+    let props: PropsDef = {};
+    let methods: MethodDef = {};
+    let wire: WireHash | undefined;
```

Comment:
L 136
```
    wire = superDef.wire || !isUndefined(wire) ? assign(create(null), superDef.wire, wire) : undefined;
```

Reply:
what is it?

Reply:
In line 136, the ternary operator condition is coercing the value of `wire`, so my suggestion was to explicitly check if `!isUndefined(wire)`